rmw_connext_shared_cpp/src/types/custom_publisher_listener.cpp

```diff
@@ -23,7 +23,7 @@
 void CustomPublisherListener::on_data_available(DDS::DataReader * reader)
 {
   DDS::PublicationBuiltinTopicDataDataReader * builtin_reader =
-    static_cast<DDS::PublicationBuiltinTopicDataDataReader *>(reader);
+    DDS::PublicationBuiltinTopicDataDataReader::narrow(reader);
```

Comment:
It looks like [narrow returns `NULL` if the cast is invalid](https://community.rti.com/kb/what-purpose-narrow-method-created-type-rtiddsgen). Mind adding a check for this case?

Reply:
Same comment on the other `narrow` call added by this PR